FirebaseCore/Sources/FIRApp.m

```diff
@@ -606,6 +610,32 @@ + (BOOL)hasSwiftRuntime {
   return hasSwiftRuntime;
 }
 
++ (NSString *)applePlatform {
+  NSString *applePlatform = @"unknown";
+
+#if TARGET_OS_IOS
+  applePlatform = @"ios";
+#endif  // TARGET_OS_IOS
+
+#if TARGET_OS_TV
+  applePlatform = @"tvos";
+#endif  // TARGET_OS_TV
+
+#if TARGET_OS_OSX
+  applePlatform = @"macos";
+#endif  // TARGET_OS_OSX
+
+#if TARGET_OS_WATCH
+  applePlatform = @"watchos";
+#endif  // TARGET_OS_WATCH
+
+#if TARGET_OS_MACCATALYST
+  applePlatform = @"maccatalyst";
```

Comment:
Please check that this define is not combined with any of the others.  We might want it first with `#elif`s

Reply:
According to the current API docs it should not. I wonder if we can test it on Travis/GHA somehow? 
<img width="789" alt="Screen Shot 2020-02-19 at 3 30 48 PM" src="https://user-images.githubusercontent.com/1841926/74873387-1938ad80-532d-11ea-8754-93cc2f9b87fc.png">

Reply:
Ok, for Catalyst app run on macOS both `TARGET_OS_IOS` and `TARGET_OS_MACCATALYST` are set. I wonder what platform we would like to log in this case? I assume `maccatalyst`. WDYT?

Reply:
I would say `maccatalyst`.

Reply:
Updated with `#elif` accordingly.